### PR TITLE
Add missing delay instruction to fake backends

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit adds the delay instruction to the target for fake backends
based on BackendV2. With BackendV2 and the Target delay isn't assumed to
be present on every backend and it has to be explictly listed. This
commit fixes this so that the compiler is aware that delay is a valid
operation (without any additional constraints) on the BackendV2 based
fake backends.

### Details and comments

### CODE REVIEW

**Review by kevinhartman — APPROVED** [review]
LGTM

**Review by HuangJunye — DISMISSED** [review]
Looks good to me too!

File: qiskit/test/mock/utils/backend_converter.py [hunk 1]
```diff
@@ -123,6 +124,9 @@ def convert_to_target(conf_dict: dict, props_dict: dict = None, defs_dict: dict
                             target[inst][(qubit,)].calibration = sched
                     else:
                         target[inst][qarg].calibration = sched
+    target.add_instruction(
+        Delay(Parameter("t")), {(bit,): None for bit in range(target.num_qubits)}
```
Comment (mtreinish) [inline qiskit/test/mock/utils/backend_converter.py:128]: I also noticed that reset is missing while working on https://github.com/Qiskit/red-queen/pull/6 so we should add that here before merging this.

Comment (mtreinish) [reply qiskit/test/mock/utils/backend_converter.py:128]: Never mind, ignore this. I was running on a fake backend that predated the reset support. Reset is properly supported already

**Review by jakelishman — APPROVED** [review]
Is this a bug in 0.20 that needs a release note, or are we good without it?

**Review by jakelishman — APPROVED** [review]

File: releasenotes/notes/delay-fake-backends-3f68c074e85d531f.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  - |
+    The :class:`~.BackendV2` based fake backends in
+    the :mod:`qiskit.providers.fake_provider` module, such as
+    ``FakeMontrealV2``, previously were missing the :class:`~.Delay` operation
+    support from their :attr:`~.BackendV2.target` attribute. This prevented
+    compiling some :class:`~.QuantumCircuit` objects that contained
+    :class:`~.Delay` instructions to these backends. This has been corrected
+    by adding the :class:`~.Delay` to the :class:`~.Target` object for each
+    fake backend to ensure that the compiler knows that :class:`~.Delay` is
+    a valid operation on the fake backends.
```
Comment (mtreinish) [inline releasenotes/notes/delay-fake-backends-3f68c074e85d531f.yaml:12]: Sigh: https://dev.azure.com/qiskit-ci/qiskit-terra/_build/results?buildId=36199&view=logs&j=abfb3e93-d1ff-5b1b-cb4c-74d6dcf536f9&t=60edc4b3-4259-59c8-ae13-dc31febe2d3d&l=7522

```suggestion
    The :class:`~.BackendV2` based fake backends in
    the :mod:`qiskit.providers.fake_provider` module, such as
    ``FakeMontrealV2``, previously were missing the :class:`~qiskit.circuit.Delay` operation
    support from their :attr:`~.BackendV2.target` attribute. This prevented
    compiling some :class:`~.QuantumCircuit` objects that contained
    :class:`~qiskit.circuit.Delay` instructions to these backends. This has been corrected
    by adding the :class:`~qiskit.circuit.Delay` to the :class:`~.Target` object for each
    fake backend to ensure that the compiler knows that :class:`~qiskit.circuit.Delay` is
    a valid operation on the fake backends.
```

**Review by jakelishman — APPROVED** [review]
Oh yeah, I always forget that `pulse` defines a bunch of objects with the same names as things in `circuit`.

**Pull request merged**